projects/bp-gallery/src/components/views/geoguessr/GeoView.tsx

```diff
@@ -0,0 +1,160 @@
+import { Box, Button, Modal, Typography } from '@mui/material';
+import { Info } from '@mui/icons-material';
+import { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  useGetAllPicturesByArchiveQuery,
+  useGetPictureGeoInfoQuery,
+} from '../../../graphql/APIConnector';
+import { useSimplifiedQueryResponseData } from '../../../graphql/queryUtils';
+import { FlatArchiveTag, FlatPictureGeoInfo } from '../../../types/additionalFlatTypes';
+import ZoomWrapper from '../picture/overlay/ZoomWrapper';
+import GeoMap from './GeoMap';
+import { useStorageState } from 'react-use-storage-state';
+import useGetPictureLink from '../../../hooks/get-pictureLink.hook';
+
+const getAllPictureIds = (archives: FlatArchiveTag[]) => {
+  const allPictureIds: string[] = archives
+    .map(archive => archive.pictures ?? [])
+    .flat()
+    .map(picture => picture.id);
+  return allPictureIds;
+};
+
+const shufflePictureIds = (pictureIds: string[], seed: number) => {
+  const newPictureIds = [...pictureIds];
+  for (let i = newPictureIds.length - 1; i > 0; i--) {
+    const j = Math.abs(Math.floor(i * Math.sin(i % seed)));
+    [newPictureIds[i], newPictureIds[j]] = [newPictureIds[j], newPictureIds[i]];
+  }
+  return newPictureIds;
+};
+
+const getTodaysPictureQueue = (pictureIds: string[]) => {
+  const pictureNumber = 10;
+  const currentDate = new Date();
+  const startDate = new Date(currentDate.getFullYear(), 0, 1);
+  const days = Math.floor((currentDate.valueOf() - startDate.valueOf()) / (24 * 60 * 60 * 1000));
+  const resultIndex = pictureIds.length % days;
+  const list = [];
+  for (let i = 0; i < pictureNumber; i++) {
+    list.push(pictureIds[(resultIndex + i) % pictureIds.length]);
+  }
+  return list;
+};
```

Comment:
Calculating today's picture queue is probably something that you could do on the backend, especially since all users get shown the same daily pictures. I could imagine that fetching, mapping and then shuffling over 20k pictures for every user could become a problem on the production server, especially because we're only using 10 of them in the end